btc-lsp/src/BtcLsp/Grpc/Data.hs

```diff
@@ -5,55 +5,19 @@
 
 module BtcLsp.Grpc.Data
   ( GRel (..),
-    PrvKey (..),
-    PubKey (..),
-    Sig (..),
     SigHeaderName (..),
     TlsCert (..),
     TlsKey (..),
-    sign,
-    verify,
   )
 where
 
 import Data.Aeson (FromJSON (..), withText)
-import Data.Coerce (coerce)
-import Data.Signable (Signable)
-import qualified Data.Signable as Signable
 import qualified Data.Text.Encoding as TE
-import Text.PrettyPrint.GenericPretty (Out (..))
 import Text.PrettyPrint.GenericPretty.Instance ()
 import Universum
-import qualified Prelude
 
 data GRel = Client | Server
 
-newtype PrvKey (rel :: GRel)
```

Comment:
Restore parametrized newtypes (with rel parameter) for PrvKey, PubKey and Sig, and use them instead of ByteString. User WItch (from, into, tryFrom, via or other combinators) instead of coerce.